feats(react-dialog): Adds dialog stories (#23963)
microsoft/fluentui: packages/react-components/react-dialog/src/stories/DialogBestPractices.md

<details>
<summary>
 Best Practices
</summary>

### Do

- Dialog boxes consist of a header (`DialogTitle`), body (`DialogBody`), and footer (`DialogActions`).
- Validate that people’s entries are acceptable before closing the dialog. Show an inline validation error near the field they must correct.
- Modal dialogs should be used very sparingly—only when it’s critical that people make a choice or provide information before they can proceed. Thee dialogs are generally used for irreversible or potentially destructive tasks. They’re typically paired with an overlay without a light dismiss.

### Don't

- Don't use more than three buttons between `DialogActions`.

> - Dialog should not open another Dialog.
> - Dialog should no be without a focusable element inside.

Reviewer: ```suggestion
- Don't open a Dialog from a Dialog
- Don't use a Dialog with no focusable elements
```
